src/main/scala/ai/privado/languageEngine/go/processor/GoProcessor.scala

```diff
@@ -43,230 +47,65 @@ class GoProcessor(
-            }
-          }
-          statsRecorder.endLastStage()
-          // Check if any of the export failed
-          if (errorMsg.toList.isEmpty)
-            Right(())
-          else
-            Left(errorMsg.toList.mkString("\n"))
-        } finally {
-          cpg.close()
-          import java.io.File
-          val cpgOutputPath = s"$sourceRepoLocation/$outputDirectoryName/$cpgOutputFileName"
-          val cpgconfig = Config()
-            .withInputPath(sourceRepoLocation)
-            .withOutputPath(cpgOutputPath)
-          val cpgFile = new File(cpgconfig.outputPath)
-          println(s"\n\nBinary file size -- ${cpgFile.length()} in Bytes - ${cpgFile.length() * 0.000001} MB\n")
-        }
-      }
+  override def applyPrivadoPasses(cpg: Cpg): List[CpgPassBase] = {
+    List(
+      {
+        if (privadoInput.assetDiscovery)
+          new JsonPropertyParserPass(cpg, s"$sourceRepoLocation/${Constants.generatedConfigFolderName}")
+        else
+          new PropertyParserPass(cpg, sourceRepoLocation, ruleCache, Language.GO, propertyFilterCache)
+      },
+      new GoYamlLinkerPass(cpg),
+      new SQLParser(cpg, sourceRepoLocation, ruleCache),
+      new SQLQueryParser(cpg),
+      new ORMParserPass(cpg, ruleCache)
+    )
+  }
 
-      case Failure(exception) =>
-        logger.error("Error while parsing the source code!")
-        logger.debug("Error : ", exception)
-        MetricHandler.setScanStatus(false)
-        Left("Error while parsing the source code: " + exception.toString)
-    }
+  override def runPrivadoTagger(cpg: Cpg, taggerCache: TaggerCache): Unit = {
+    cpg.runTagger(ruleCache, taggerCache, privadoInput, dataFlowCache, appCache, databaseDetailsCache, statsRecorder)
   }
 
-  /** Create cpg using Go Language
-    *
-    * @param sourceRepoLocation
-    * @param lang
-    * @return
-    */
-  def createGoCpg(): Either[String, Unit] = {
-    statsRecorder.justLogMessage("Processing source code using GoLang engine")
+  override def processCpg(): Either[String, CpgWithOutputMap] = {
+    statsRecorder.justLogMessage("Processing source code using GO engine")
     statsRecorder.initiateNewStage("Base source processing")
 
-    // Converting path to absolute path, we may need that same as JS
-    val absoluteSourceLocation = File(sourceRepoLocation).path.toAbsolutePath
-    val cpgOutputPath          = s"$sourceRepoLocation/$outputDirectoryName/$cpgOutputFileName"
-    val excludeFileRegex       = ruleCache.getRule.exclusions.flatMap(rule => rule.patterns).mkString("|")
+    createCpgFolder(sourceRepoLocation)
 
-    // Create the .privado folder if not present
-    createCpgFolder(sourceRepoLocation);
+    val cpgOutputPath = s"$sourceRepoLocation/$outputDirectoryName/$cpgOutputFileName"
 
     val cpgconfig = Config()
       .withInputPath(sourceRepoLocation)
       .withOutputPath(cpgOutputPath)
-      .withIgnoredFilesRegex(excludeFileRegex)
+      .withIgnoredFilesRegex(ruleCache.getExclusionRegex)
```

Comment:
If you remember earlier there was some issue using the `getExclusionRegex` in Go

Reply:
instead of this we can also do this 
`val excludeFileRegex = ruleCache.getRule.exclusions.flatMap(rule => rule.patterns).mkString("|")`